contracts/Token.sol

```diff
@@ -152,7 +181,33 @@ contract Token is Initializable, IToken, IERC20 {
     return allowances[owner][spender];
   }
 
-  // ------------------------ Public but app functions.
+  /// --- ERC1404 functions.
+
+  function detectTransferRestriction (address owner, address recipient, uint256) public view returns (uint8) {
```

Comment:
@afmfe-iul  This is the transfer restriction detection method. Based on its input, it can return an error code or zero if everything is good.

Note that this function shall be called from the front-end wallet in the next iteration, so the results are consistent with the ERC1404 implementation.
Basically when a user would push the "Transfer" button in the front-end, we would run the detection function, and if the return code isn't zero we would display a message explaining that a restriction prevents the transfer to be initiated. Since the function is a `view` (not a transaction), it can be called for free.